pkg/engine/engine.go

```diff
@@ -239,8 +240,17 @@ func (e *Engine) QueueRun(request *api.RunRequest, sources *api.UnpackedSources)
 				Created: time.Now().UTC(),
 			},
 		},
-		CreatedBy: task.CreatedBy(request.CreatedBy),
-	})
+		CreatedBy: cby,
+	}
+
+	// Remove existing tasks from same branch end repo before pushing a new task
+	err := e.queue.RemoveExisting(newTask.CreatedBy.Branch, newTask.CreatedBy.Repo)
```

Comment:
What happens if, as a developer working on my machine, I try to submit multiple jobs to my daemon?
I guess the `cby.Branch` and `cby.Repo` are empty strings and my jobs will cancel each others.

Reply:
Correct - I forgot to add a check that `branch` and `repo` should not be empty strings